### PR TITLE
Openbook DCA Program on Mainnet

### PR DESCRIPTION
**Pull request by Eliascm17**
- Get Openbook DCA program to execute a Dollar Cost Averaging investment strategy for a given Openbook market on mainnet.

### CODE REVIEW

File: openbook_dca/programs/openbook_dca/src/instructions/swap.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+};
+
+#[derive(Accounts)]
+pub struct Swap<'info> {
+    #[account(
+        mut,
+        associated_token::mint = dca.coin_mint,
+        associated_token::authority = dca.authority
+    )]
+    pub authority_coin_vault: Box<Account<'info, TokenAccount>>,
```
Comment (segfaultdoc) [inline openbook_dca/programs/openbook_dca/src/instructions/swap.rs:30]: this the coin we DCA into?

File: openbook_dca/programs/openbook_dca/src/instructions/swap.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+    #[account(
+        mut,
+        associated_token::mint = dca.pc_mint,
+        associated_token::authority = dca.authority
+    )]
+    pub authority_pc_vault: Box<Account<'info, TokenAccount>>,
```
Comment (segfaultdoc) [inline openbook_dca/programs/openbook_dca/src/instructions/swap.rs:37]: assuming DCA from?

File: openbook_dca/programs/openbook_dca/src/instructions/swap.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    pub dca_pc_vault: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        signer,
+        address = dca_thread.pubkey(),
+        constraint = dca_thread.authority == dca.authority
+    )]
+    pub dca_thread: Box<Account<'info, Thread>>,
+
+    #[account(address = sysvar::rent::ID)]
```
Comment (segfaultdoc) [inline openbook_dca/programs/openbook_dca/src/instructions/swap.rs:73]: dnt need this and the address checks below, just being safe-r?

File: openbook_dca/programs/openbook_dca/src/instructions/swap.rs [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+                SEED_DCA,
+                dca.authority.as_ref(),
+                dca.market.as_ref(),
+                &[bump],
+            ]],
+        ),
+        dca.swap_amount,
+    )?;
+
+    // place order on serum dex
```
Comment (segfaultdoc) [inline openbook_dca/programs/openbook_dca/src/instructions/swap.rs:127]: serum or openbook?

File: openbook_dca/programs/openbook_dca/src/instructions/swap.rs (hunk 1 above)
Comment (Eliascm17) [reply openbook_dca/programs/openbook_dca/src/instructions/swap.rs:30]: yes

File: openbook_dca/programs/openbook_dca/src/instructions/swap.rs (hunk 2 above)
Comment (Eliascm17) [reply openbook_dca/programs/openbook_dca/src/instructions/swap.rs:37]: yes

**Pull request merged**